crates/typst/src/visualize/image/mod.rs

```diff
@@ -190,6 +190,22 @@ fn layout_image(
         Smart::Auto => determine_format(elem.path().as_str(), data).at(span)?,
     };
 
+    // Warn the user if the image contains a foreign object. Not perfect
+    // because the svg could also be encoded, but that's an edge case.
+    if format == ImageFormat::Vector(VectorFormat::Svg) {
+        let has_foreign_object =
+            data.as_str().is_some_and(|s| s.contains("<foreignObject"));
+
+        if has_foreign_object {
```

Comment:
This might be not an issue when exporting to svg.

Reply:
Fair enough, but since we don't have export-specific lints yet, that's the only thing we can do. And it says _might render_ incorrectly, anyway.

Reply:
I mean, for example, people using touying usually embed gifs by this approach but they will get an warning that cannot escape.

Reply:
Yeah, that's annoying... But hopefully it won't be an issue once we have warning surpression.